strongbox-event-api/src/main/java/org/carlspring/strongbox/event/artifact/ArtifactEventListenerRegistry.java

```diff
@@ -110,7 +152,9 @@ public void dispatchArtifactDownloadingEvent(String storageId, String repository
         dispatchEvent(event);
     }
 
-    public void dispatchArtifactDownloadedEvent(String storageId, String repositoryId, String path)
+    public void dispatchArtifactDownloadedEvent(String storageId,
```

Comment:
Am I correctly understood that he would be called twice?
once in `MavenArtifactController` and another in `ProxyRepositoryArtifactResolver`
Maybe it will be better to use different event type for Proxy Repositories?

Reply:
@carlspring  and why do we have this events only in `MavenArtifactController`, they will not work for Nuget in this case.

Reply:
> Am I correctly understood that he would be called twice?
once in MavenArtifactController and another in ProxyRepositoryArtifactResolver
Maybe it will be better to use different event type for Proxy Repositories?

It looks so, I haven't noticed it. @carlspring , when should we dispatch artifact downloaded event ? When the artifact is downloaded from the proxy repository ? Or, when the artifact has been requested and flushed from the Controller ? 

Thanks, @sbespalov

Reply:
Hi,

Thanks for spotting this, @sbespalov!

@fuss86: I believe the better option would be #2 -- after it's flushed. What do you guys think?

Reply:
Ok, I'll add one new event when the proxied file has been downloaded from remote repository :)